### DISCUSSION TITLE
NMEA0183 (RS422)

### DISCUSSION THREAD

**Discussion by peterantypas** [peterantypas/maiana#69] [Q&A] [answered]
Hello all,

I'm trying to wrap up NMEA0183 support (several people are waiting for this), but I don't have any NME0183 devices. I took this scope shot:

![NMEA0183-Uart+RS422](https://user-images.githubusercontent.com/1565933/124296637-092f0180-db0f-11eb-8339-a3352675aeb2.png)

Blue is the UART TX signal from the unit. White is the differential from the output of the line driver IC (TI DS8921M). Does this look correct?

Thanks!

**Comment by gah67** [comment]
Hmmm - I'm not sure as I think RS-422 is reversed polarity vs UART?? BUT, how have you connected A & B to Ch1 & 2? TBH I find it easier with the 2 individual traces rather than the difference 

For a '1' or Mark B>A and '0' or space A>B 

Either way - just reverse the labelling if needed! 

NMEA0183 just uses RS-422 (EIA-422) at the hardware level - A cheapy $3 USB/RS-485 works fine

**Reply by peterantypas** [reply]
Ch1 is on OD+, Ch2 is on OD- of the IC. Good point about reversing the signals if unsure. Either way, they're supposed to go to an optocoupler, so no harm done (hopefully).

**Comment by peterantypas** [comment]
Also, regarding USB: The breakout box has both USB and NMEA0183 (RS422), with the latter being output only. I want to keep the USB interface on all breakouts, including the NMEA2000 (which is in progress).

**Reply by peterantypas** [reply]
This post suggests that the polarity of CMOS/TTL matches RS422:

https://www.bb-elec.com/Learning-Center/All-White-Papers/Serial/%E2%80%A2-Polarities-for-Differential-Pair-Signals-(RS-422.aspx

**Comment by astuder** [comment]
For the electrical side of NMEA 0183 I'm a big fan of this document:
https://actisense.com/wp-content/uploads/2020/01/NMEA-0183-Information-sheet-issue-4-1-1.pdf
It agrees with gah67 regarding polarity.

**Comment by peterantypas** [comment]
OK, here is the truth table from the data sheet:

![Screenshot from 2021-07-02 11-31-40](https://user-images.githubusercontent.com/1565933/124315683-2670c980-db29-11eb-86ac-a6d983ce439e.png)

Are you saying that DO+ is actually "NMEA-"?

**Comment by gah67** [comment]
I think it's all ok - all the docs posted all agree - it's the labelling that changed over time. The OD+ & OD- correspond to RS-422 B & A respectively 

My brain (having done a fair bit of 422 & 485 work in the past) always thinks of Ch1 & Ch2 on a scope as A & B rather than the other way round so I am just used to seeing it inverted 
Also RS232 is inverted logic as well so it really just follows but then some don't like the inversion and change all the labelling to suit :)

**Comment by peterantypas** [comment]
Here is the single ended output:

![SDS00011](https://user-images.githubusercontent.com/1565933/124316277-3806a100-db2a-11eb-81ed-ca40a69b9d46.png)

Yellow = UART
Pink = OD+
Blue = OD-

So, the question now is which signal is "NMEA+"?

**Comment by gah67** [comment, marked as answer]
If the yellow trace is TTL then the purple trace is TX+ in NMEA0183 terms - this is the single wire when used with a single wire listener and connected to RX+ on a differential listener 

(It is B when connected to RS-422 terminals)

**Reply by peterantypas** [reply]
Yes, yellow is CMOS/TTL output. I can make the front panel label now ;)

Thanks!
